orm_lib/src/TransactionImpl.cc

```diff
@@ -148,7 +149,68 @@ void TransactionImpl::execSqlInLoop(
         exceptCallback(exceptPtr);
     }
 }
-
+void TransactionImpl::commit(std::function<void(bool)> callback)
+{
+    auto thisPtr = shared_from_this();
+    if (callback)
+    {
+        commitCallback_ = std::move(callback);
+    }
+    loop_->runInLoop([thisPtr]() {
+        if (thisPtr->isCommitedOrRolledback_)
+            return;
+        if (thisPtr->isWorking_)
+        {
+            // push sql cmd to buffer;
+            auto cmdPtr = std::make_shared<SqlCmd>();
+            cmdPtr->sql_ = "commit";
+            cmdPtr->parametersNumber_ = 0;
+            cmdPtr->callback_ = [thisPtr](const Result &) {
+                LOG_DEBUG << "Transaction commit!";
+                thisPtr->isCommitedOrRolledback_ = true;
+                if (thisPtr->commitCallback_)
+                {
+                    thisPtr->commitCallback_(true);
+                }
+            };
+            cmdPtr->exceptionCallback_ = [thisPtr](const std::exception_ptr &) {
+                thisPtr->isCommitedOrRolledback_ = true;
+                if (thisPtr->commitCallback_)
+                {
+                    thisPtr->commitCallback_(false);
+                }
+                LOG_ERROR << "Transaction commit error";
+            };
+            cmdPtr->isRollbackOrCommitCmd_ = true;
+            thisPtr->sqlCmdBuffer_.push_back(std::move(cmdPtr));
+            return;
+        }
+        thisPtr->isWorking_ = true;
+        thisPtr->thisPtr_ = thisPtr;
+        thisPtr->connectionPtr_->execSql(
+            "commit",
+            0,
+            {},
+            {},
+            {},
+            [thisPtr](const Result &) {
+                LOG_TRACE << "Transaction commit!";
+                thisPtr->isCommitedOrRolledback_ = true;
+                if (thisPtr->commitCallback_)
+                {
+                    thisPtr->commitCallback_(true);
+                }
+            },
+            [thisPtr](const std::exception_ptr &) {
```

Comment:
Since this is the last place we used `thisPtr`. Can move move it to avid a copy?